lib/index.js

```diff
@@ -208,11 +203,53 @@ var processRef = function (property, objectName, props, key, required) {
   fillRequired(props, key, required);
 };
 
+var isPolymorphic = function (def) {
+  return def['allOf'];
+}
+
+/*
+ * Merges members of 'allOf' into a single 'properties' object,
+ * and a single 'required' array, and assigns these to the parent definition.
+ *
+ * NOTE: needs more thorough testing!
+ */
+var processPolymorphic = function (definition) {
+  var mergedProps = {};
+  var mergedReqs = new Set();
+  definition['allOf'].forEach(function(polyDef) {
+    if(polyDef['$ref']) {
+      var refRegExp = /^#\/definitions\/(\w*)$/;
+      var refString = polyDef['$ref'];
+      var propType = refString.match(refRegExp)[1];
+      var object = definitions[propType];
+      Object.assign(mergedProps, object['properties']);
+      if (object['required']) {
+        object['required'].forEach(function(req) {
+          mergedReqs.add(req);
+        });
+      }
+    }else {
+      Object.assign(mergedProps, polyDef['properties']);
+      if (polyDef['required']) {
+        polyDef['required'].forEach(function(req) {
+          mergedReqs.add(req);
+        });
+      }
+    }
+  });
+  if (!_.isEmpty(mergedProps)) definition['properties'] = mergedProps;
+  if (!_.isEmpty(mergedReqs)) definition['required'] = Array.from(mergedReqs);
+  delete definition['allOf'];
+}
+
 var getSchema = function (objectName, fullObject) {
   var props = {};
   var required = fullObject.required || [];
```

Comment:
Line 247 (`var required = fullObject.required || [];`) should be moved to line 251 (i.e. after processPolymorphic runs if the definition is polymorphic) otherwise we're not adding in the merged required fields.